superset-frontend/src/filters/components/GroupBy/GroupByFilterPlugin.tsx

```diff
@@ -68,7 +68,15 @@ export default function PluginFilterGroupBy(props: PluginFilterGroupByProps) {
     // so we can process it like this `JSON.stringify` or start to use `Immer`
   }, [JSON.stringify(defaultValue), multiSelect]);
 
-  const columns = data || [];
+  const columns = data
+    ? formData.groupby && formData.groupby[0] && formData.groupby[0].length
```

Comment:
I think we can do it shorter:
```
formData?.groupby?.[0]?.length
```

Reply:
This is slightly heavy to read. Could we break this up a bit, something like
```js
const groupby = ensureIsArray(formData.groupby);
const columns = data ? data.filter(dataItem => groupby.includes(dataItem.columnName) : [];
```